bootstrap/internal/controller/infraenv_controller_test.go

```diff
@@ -156,4 +160,169 @@ var _ = Describe("InfraEnv Controller", func() {
 			})
 		})
 	})
+	Context("InfraEnv Controller uses internal URL for the ISO", func() {
+		ctx := context.Background()
+		var controllerReconciler *InfraEnvReconciler
+		var k8sClient client.Client
+		var abc *bootstrapv1alpha1.AgentBootstrapConfig
+
+		BeforeEach(func() {
+			k8sClient = fakeclient.NewClientBuilder().WithScheme(testScheme).
+				WithStatusSubresource(&bootstrapv1alpha1.AgentBootstrapConfig{}).
+				Build()
+			Expect(k8sClient).NotTo(BeNil())
+
+			controllerReconciler = &InfraEnvReconciler{
+				Client: k8sClient,
+				Scheme: k8sClient.Scheme(),
+				Config: InfraEnvControllerConfig{
+					UseInternalImageURL:   true,
+					ImageServiceNamespace: namespace,
```

Comment:
Can we change it to simulate a more realistic scenario and change it to a different namespace?

Reply:
Changed it to a different namespace, thanks!!